keanu-python/keanu/net.py

```diff
@@ -15,10 +18,12 @@
 
 class BayesNet(JavaObjectWrapper):
 
-    def __init__(self, vertices: Iterable[Any]) -> None:
-        java_vertices = k.to_java_object_list(vertices)
-
-        super(BayesNet, self).__init__(k.jvm_view().BayesianNetwork(java_vertices))
+    def __init__(self, vertices_or_java_object: Union[Iterable[Any], JavaObject]) -> None:
```

Comment:
@icantsayrural @migwellian I'm currently changing this constructor so I can call `BayesNet(java_bayes_net)` in `Sequence.to_bayes_net()` but is there a better way of doing this?

Reply:
I don't think so, this is how we do it usually. It is a bit weird here since `BayesNet` is a public class and we'd usually want to hide the java part as much as possible (the parameter `vertices_or_java_object`)... but I don't think there is a better solution.

In the past, I've tried creating a separate `_BayesNet` class that just takes the java object in the constructor and having `BayesNet` extend that. But this just adds unnecessary complexity for so little gain.

Reply:
Okay I've gone with a slightly different approach of constructing a JavaObjectWrapper directly and updating the instance's `__class__` from within the `to_bayes_net()` method.

Reply:
:+1: